src/qibojit/custom_operators/__init__.py

```diff
@@ -290,6 +290,18 @@ def cpu_cast(self, x, dtype='DTYPECPX'):
         except AttributeError:
             return super().cpu_cast(x, dtype)
 
+    def cpu_fallback(self, func, *args):
+        """Executes a function on CPU if the default devices raises OOM."""
+        try:
+            return func(*args)
+        except self.oom_error: # pragma: no cover
+            # case not covered by GitHub workflows because it requires OOM
+            log.warn("Falling back to CPU because the GPU is out-of-memory.")
```

Comment:
Given that this is usually a part of larger computation part of which may run successfully on GPU (for example on circuit simulation the gates are applied on GPU and only the measurement falls back to CPU), it may be usefull to say which function is used in the fallback. One option is the following:
```suggestion
            log.warn(f"Falling back to CPU for {func.__name__} because the GPU is out-of-memory.")
```
What do you think?

Reply:
Looks good, I'll give it a try.